### ISSUE TITLE
FSE: Decouple Navigation Block From Themes

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#34793]
Our goal with the new FSE editing flows is to make sure folks are able to do everything they need within the Block Editor

Let's update the navigation block to show an autogenerated menu the reflects a site's current pages. If a page is added or renamed, it'll reflect in this menu automatically. 

In terms of technical tasks, this means we should decouple the rendering call from Theme logic:
https://github.com/Automattic/wp-calypso/blob/e7c6389499574b6841fb1c21aa479f4c76669dc6/apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/navigation-menu/index.php#L91

Basically this should work regardless of if someone has made a menu in the customizer or not, and should work regardless of what theme is selected.

**Comment by apeatling** [comment]
Fixed with https://github.com/Automattic/wp-calypso/pull/34796
